pkg/controller/core/resourceflavor_controller.go

```diff
@@ -77,11 +77,11 @@ func (r *ResourceFlavorReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling ResourceFlavor")
 
-	if flavor.ObjectMeta.DeletionTimestamp.IsZero() {
+	if flavor.DeletionTimestamp.IsZero() {
 		// Although we'll add the finalizer via webhook mutation now, this is still useful
 		// as a fallback.
-		if !controllerutil.ContainsFinalizer(&flavor, kueue.ResourceInUseFinalizerName) {
-			controllerutil.AddFinalizer(&flavor, kueue.ResourceInUseFinalizerName)
+		if controllerutil.AddFinalizer(&flavor, kueue.ResourceInUseFinalizerName) {
+			log.V(5).Info("Add finalizer")
```

Comment:
`Added finalizer`, and it should be after the operation (Update) succeeded.

Reply:
done